### PR TITLE
Re-issue ROAs and CA certificates on key activation. (#509)

### PR DESCRIPTION
**Pull request by timbru**

### CODE REVIEW

**Comment by timbru** [comment]
Now also featuring a test in `functional.rs` that verifies that a ROA and certificate are in fact re-issued. Before the test was just checking key status - I can't really say why it was not checking this. I assumed that it was I guess - which is why it was missed in the refactoring.

File: src/daemon/ca/rc.rs [hunk 1]
```diff
@@ -450,18 +450,42 @@ impl ResourceClass {
...
 
-        Ok(vec![self.key_state.keyroll_activate(
-            self.name.clone(),
-            self.parent_rc_name.clone(),
-            signer,
-        )?])
+                let mut cert_updates = ChildCertificateUpdates::default();
+                for issued in self.certificates.iter() {
+                    // re-issue
+                    let re_issued = self.re_issue(issued, None, new_key, None, issuance_timing, signer)?;
```
Comment (ximon18) [inline src/daemon/ca/rc.rs:476]: If `self.re_issue()` succeeds on one loop iteration but fails on the next this fn will abort, but in that case the event for the update that happened will not be returned. Is this okay?

File: tests/migrate_repository.rs [hunk 2]
```diff
@@ -403,6 +415,13 @@ async fn migrate_repository() {
                 .await
             );
         }
+
+        // Expect that CA3 publishes nothing in the embedded repo
```
Comment (ximon18) [inline tests/migrate_repository.rs:419]: This comment says CA3 but the following code says CA1, is this correct?

**Review by ximon18 — APPROVED** [review]
The logic of the test is beyond my ability to confirm but I can see that you are indeed testing more than you were before.

File: src/daemon/ca/rc.rs (hunk 1 above)
Comment (timbru) [reply src/daemon/ca/rc.rs:476]: That is okay, none of this modifies the actual CA. This is creating a number of events which will only be applied if everything was successful - including triggered pre-save effects (create new mft and crl in particular).

File: tests/migrate_repository.rs (hunk 2 above)
Comment (timbru) [reply tests/migrate_repository.rs:419]: indeed copy/paste error from previous test. This will be reviewed as part of issue #480

**Pull request merged**